src/circuit/gadget/ecc/chip/constants.rs

```diff
@@ -0,0 +1,273 @@
+use arrayvec::ArrayVec;
+use group::{
+    ff::{Field, PrimeField},
+    Curve,
+};
+use halo2::arithmetic::lagrange_interpolate;
+use pasta_curves::{
+    arithmetic::{CurveAffine, FieldExt},
+    pallas,
+};
+
+/// Window size for fixed-base scalar multiplication
+pub const FIXED_BASE_WINDOW_SIZE: usize = 3;
+
+/// $2^{`FIXED_BASE_WINDOW_SIZE`}$
+pub const H: usize = 1 << FIXED_BASE_WINDOW_SIZE;
+
+/// Number of windows for a full-width scalar
+pub const NUM_WINDOWS: usize =
+    (pallas::Scalar::NUM_BITS as usize + FIXED_BASE_WINDOW_SIZE - 1) / FIXED_BASE_WINDOW_SIZE;
+
+/// Number of windows for a short signed scalar
+pub const NUM_WINDOWS_SHORT: usize =
+    (L_SCALAR_SHORT + FIXED_BASE_WINDOW_SIZE - 1) / FIXED_BASE_WINDOW_SIZE;
+
+/// $\ell_\mathsf{value}$
+/// Number of bits in an unsigned short scalar.
+pub(crate) const L_SCALAR_SHORT: usize = 64;
+
+/// The Pallas scalar field modulus is $q = 2^{254} + \mathsf{t_q}$.
+/// <https://github.com/zcash/pasta>
+pub(crate) const T_Q: u128 = 45560315531506369815346746415080538113;
+
+/// The Pallas base field modulus is $p = 2^{254} + \mathsf{t_p}$.
+/// <https://github.com/zcash/pasta>
+pub(crate) const T_P: u128 = 45560315531419706090280762371685220353;
```

Comment:
I checked these constants (`T_Q` and `T_P`).

Reply:
However, they could be imported from `constants.rs`, I think.

Reply:
`src/constants.rs` will be in the `orchard` crate, but `ecc::chip::constants` will be in the `halo2_gadgets` crate.

Reply:
OK, then the former should import these constants from the latter.